openhtf/plugs/acroname/power_module_plug.py

```diff
@@ -0,0 +1,306 @@
+ """No EtherStem module is found on the network."""
+
+class NoMatchingEtherStemFoundError(Exception):
+ """No EtherStem module matching the target EtherStem is found on the network."""
+
+class ConnectingToEtherStemError(Exception):
+ """Error connecting to EtherStem."""
+
+class ConnectingToPowerModuleThroughEtherStemError(Exception):
+ """Error connecting power module through EtherStem."""
+
+class ConfigurePowerSupplyError(Exception):
+ """Error configuring power supply."""
+
+class TurnOnPowerSupplyError(Exception):
+ """Error turning on power supply."""
+
+class ChangeVoltageError(Exception):
+ """Error changing voltage."""
+
+class ChangeCurrentLimitError(Exception):
+ """Error changing current limit."""
+
+class TurnOffPowerSupplyError(Exception):
+ """Error turning off power supply."""
+
+ 
+
+
+class PowerSupplyControl(plugs.BasePlug):   # pylint: disable=no-init
+  """Acronmae power module control plug."""
+  
+
+  def __init__(self):
+    config = conf.Config()
+    self.serial_number = config.serial_number_of_EtherStem
+    self.voltage = config.default_power_module_voltage_output
+    self.current_limit = config.default_power_module_current_limit
+    self.MTMPowerModuleAddress = 6
+    self.routerAddress = 4
+    self.EtherStem = MTMEtherStem()
+    self.power_module = MTMPM1()
+
+  def DiscoverAndConnectModule(self):
+    """
+    Discover all EtherStems on network.
+    Connect the one with the serial number specified in config yaml file.
+    The connection from server to EtherStem is Ethernet via TCPIP.
+    """
+    # codes below is to discover EtherStem on network
+    stem_list = discover.find_all(Spec.TCPIP)
+    EtherStemFound = False
+    if stem_list is not []:
+      for stem in stem_list:
+        print "stem: "+str(stem)
+        if stem.serial_number == self.serial_number:
+          stem_index = stem_list.index(stem)
+          spec = stem_list[stem_index]
+          EtherStemFound = True
+          break
+      if not EtherStemFound:
+        _LOG.info('No Matching EtherStem is found.')
+        raise NoMatchingEtherStemFoundError
+    else:
+      _LOG.info('None EtherStem is found.')
+      raise NoneEtherStemFoundError
+    
+    # codes below is to connect EtherStem
+    _LOG.info("Connecting to EtherStem with serial number:0x%X" %self.serial_number)
+    print("Connecting to EtherStem with serial number: 0x%X" % self.serial_number)
+    print "spec = "+str(spec)
+    res = self.EtherStem.connect_from_spec(spec)
+    print("Result of Connecting to EtherStem: " +str(res))
+    if res != 0:
+      _LOG.info("Error of connecting to EtherStem:"+str(res))
+      raise ConnectingToEtherStemError
+    else:
+      _LOG.info("Connecting to EtherStem Succeed.")
+
+    # codes below is to connect Power Module through EtherStem
+    res_list=[]
+
+    res = self.EtherStem.i2c[0].setPullup(1)
+    res_list.append(res)
+    _LOG.info("Result of setting link stem pullups:"+str(res))
```

Comment:
do we want to continue doing stuff to the EtherStem if stuff fails earlier? If not, this might be better:
```py
def CheckReturnCode(code, action_str):
  if not code:
    _LOG.info("Result of %s: %d", action_str, code)
  else:
    _LOG.error("Error connecting to power module through EtherStem while attempting to: %s", action_str)
    raise ConnectingToPowerModuleThroughEtherStemError()
```

```py
CheckReturnCode(self.EtherStem.i2c[0].setPullup(1), "setting link stem pullups")
...
_LOG.info("Success...")
```